src/jit/morph.cpp

```diff
@@ -7751,7 +7749,15 @@ GenTree* Compiler::fgMorphTailCallViaHelpers(GenTreeCall* call, CORINFO_TAILCALL
             target = fgGetDirectCallTargetAddress(call);
         }
 
-        call->gtCallArgs = gtPrependNewCallArg(target, call->gtCallArgs);
+        // Insert target as last arg
+        GenTreeCall::Use** newArgSlot = &call->gtCallArgs;
+        while (*newArgSlot != nullptr)
+        {
+            newArgSlot = &(*newArgSlot)->NextRef();
+        }
+
+        *newArgSlot = new (this, CMK_ASTNode) GenTreeCall::Use(target, nullptr);
```

Comment:
`new (this, CMK_ASTNode) GenTreeCall::Use(target, nullptr)` -> `gtNewCallArgs(target)`